src/api/app/controllers/status/checks_controller.rb

```diff
@@ -69,5 +69,11 @@ def set_xml_check
   def xml_hash
     result = (Xmlhash.parse(request.body.read) || {}).with_indifferent_access
     result.slice(:url, :state, :short_description, :name)
+    result.each do |key, value|
+      # Xmlhash converts empty elements to Hash
+      result[key] = '' if value.is_a?(Hash)
+    end
```

Comment:
hmm, I'm not code climate - but you have pretty much the same code now in 2 places 

And I'm afraid Xmlhash is just not meant to be used this way - you're supposed to xmlhash.parse(body).value('url') - and then xmlhash would make sure to return a string itself.

This casting forth and back creates the problem to begin with. 

So I would recommend 
```
result = HashWithIndifferentAccess.new
parsed_body = Xmlhash.parse(request.body.read) 
%w(url state short_description name).each { |key| result[key] = parsed_body.value(key) }
result
```

This would make the whole thing look less like a workaround

Reply:
You could also just use ``Hash.from_xml`` from ActiveSupport.

```
Hash.from_xml("<bar><name></name><bar>").with_indifferent_access
# -> {"bar"=>{"name"=>nil}}

Hash.from_xml("").with_indifferent_access
# -> nil
```

However, this will throw an exception if the XML is invalid, so you would need to catch the exception. But I would prefer that we use this method in the future and try to get rid of Xmlhash.

Reply:
Good luck with that - you will have even more fun casting different members, because:

```
 Hash.from_xml("<bar><name>Dude</name><name>Other Dude</name></bar>")['bar']['name'].each { |i| puts i }
Dude
Other Dude

Hash.from_xml("<bar><name>Only one Dude</name></bar>")['bar']['name'].each { |i| puts i }
Traceback (most recent call last):
        1: from (irb):16
NoMethodError (undefined method `each' for "Only one Dude":String)
```

It's not like Xmlhash wasn't created due to a real need. Because this is what xmlhash's api 
offers:

```
Xmlhash.parse('<bar><name>Dude</name><name>Other Dude</name></bar>').elements('name') { |i| puts i }
Dude
Other Dude

Xmlhash.parse('<bar><name>Only one Dude</name></bar>').elements('name') { |i| puts i }
Only one Dude
```

Reply:
And on top of that: ./core_ext/hash/conversions.rb looks *super slow*